src/python/pants/pantsd/watchman.py

```diff
@@ -28,24 +28,23 @@ class WatchmanCrash(Exception):
 
   EventHandler = namedtuple('EventHandler', ['name', 'metadata', 'callback'])
 
-  def __init__(self, watchman_path, work_dir, log_level='1', startup_timeout=STARTUP_TIMEOUT_SECONDS,
-               timeout=SOCKET_TIMEOUT_SECONDS, socket_path_override=None, metadata_base_dir=None):
+  def __init__(self, watchman_path, metadata_base_dir, log_level='1', startup_timeout=STARTUP_TIMEOUT_SECONDS,
+               timeout=SOCKET_TIMEOUT_SECONDS, socket_path_override=None):
     """
     :param str watchman_path: The path to the watchman binary.
-    :param str work_dir: The path to the pants work dir.
+    :param str metadata_base_dir: The metadata base dir for `ProcessMetadataManager`.
     :param float startup_timeout: The timeout for the initial `watch-project` query (in seconds).
     :param float timeout: The watchman socket timeout for all subsequent queries (in seconds).
     :param str log_level: The watchman log level. Watchman has 3 log levels: '0' for no logging,
                           '1' for standard logging and '2' for verbose logging.
     :param str socket_path_override: The overridden target path of the watchman socket, if any.
-    :param str metadata_base_dir: The overriden metadata base dir for `ProcessMetadataManager`.
     """
     super(Watchman, self).__init__(name='watchman',
                                    process_name='watchman',
                                    socket_type=str,
                                    metadata_base_dir=metadata_base_dir)
     self._watchman_path = self._normalize_watchman_path(watchman_path)
-    self._watchman_work_dir = os.path.join(work_dir, self.name)
+    self._watchman_work_dir = os.path.join(metadata_base_dir, self.name)
     self._log_level = log_level
```

Comment:
this should probably continue to use the workdir vs `.pids` for this.. since it's where e.g. logs, state files, unix sockets, etc are kept. unless we intend to repurpose `.pids` to become the "work dir for subprocesses".

did this somehow affect repl?

Reply:
> did this somehow affect repl?

It affected the ease of testing: see the description. Basically, the `@ensure_daemon` context manager doesn't know which workspaces have been by the various pants runs in an integration test. But if it's able to use the daemon by setting and controlling "only" the `subprocess_dir`, it is able to clean up all launched processes that way.

I'm not sure whether this is a fundamental split that would be useful, but on the surface, it makes sense to me: subprocess might have logs, pids, etc... and the lifecycle of the pid and the log is the same. So moving pantsd's own log next to its pid file would also make sense probably.

Reply:
discussed offline, but it would be good to add and call a ~`ProcessManager.get_process_workdir()` method here to key into a subprocess-specific workdir (e.g. `.pids/watchman/workdir`) to keep the existing file-based metadata separate from things like logs, sockets, state files etc.